trestle/core/utils.py

```diff
@@ -140,8 +140,23 @@ def get_origin(field_type: Type[Any]) -> Optional[Type[Any]]:
     from various python versions.
     """
     # This executes a fallback that allows a list to be generated from a constrained list.
-    return typing_extensions.get_origin(field_type) or getattr(field_type, '__origin__',
-                                                               None) or type(getattr(field_type, '__root__', None))
+    return typing_extensions.get_origin(field_type) or getattr(field_type, '__origin__', None)
+
+
+def _get_model_field_info(field_type):
+    """Need special handling for pydantic wrapped __root__ objects."""
+    # oscal_read of roles.json yields pydantic.Roles model with __root__ containing list of Role
+    root = None
+    root_type = None
+    singular_type = None
+    if hasattr(field_type, '__fields__'):
+        fields = field_type.__fields__
+        if '__root__' in fields:
+            root = fields['__root__']
+            singular_type = root.type_
+            if hasattr(root, 'outer_type_') and hasattr(root.outer_type_, '_name'):
+                root_type = root.outer_type_._name
```

Comment:
it might be lazier to wrap this in a try , except block (rather than 3.5 nested if's) That difference is purely stylistic in nature (e..g fix if you want).